components/automate-gateway/gateway/middleware/authv2/authv2.go

```diff
@@ -106,21 +93,21 @@ func (c *client) IsAuthorized(ctx context.Context, subjects []string, resource,
 	})
 	if err != nil {
 		if status.Convert(err).Code() == codes.FailedPrecondition {
-			return nil, err
+			return nil, false, err
 		}
 		log.WithError(err).Error("error authorizing request")
-		return nil, status.Errorf(codes.PermissionDenied,
-			"error authorizing action %q on resource %q for members %q: %s",
+		// something went wrong in some unexpected way -- we'll "fail shut", i.e., deny authorization
+		return nil, false, errors.Errorf("error authorizing action %q on resource %q for members %q: %s",
 			action, resource, subjects, err.Error())
 	}
 	projects := filteredResp.Projects
 
-	return &resp{
-		ctx:        auth_context.NewContext(ctx, subjects, projects, resource, action),
-		authorized: len(projects) != 0,
-	}, nil
+	return auth_context.NewContext(ctx, subjects, projects, resource, action),
```

Comment:
Would appreciate a comment here (or just above the function signature above) explaining the return values, most importantly that the boolean being returned indicates authorized or not, and that equates to the presence of one or more projects.